rfc/src/rfcs/0008-kani-version.md

```diff
@@ -0,0 +1,109 @@
+- **Feature Name:** Print Kani version (`kani-version`)
+- **Feature Request Issue:** <https://github.com/model-checking/kani/issues/2570>
+- **RFC PR:** <https://github.com/model-checking/kani/issues/2571>
+- **Status:** Under Review
+- **Version:** 0
+
+-------------------
+
+## Summary
+
+Print the version of Kani at the beginning of a run.
+
+## User Impact
+
+Many programs print their version at the beginning of a run.
+The version of a program communicates the state of the software at a given point (e.g., features that are available or performance on particular problems).
+At present, Kani does not print its version, but it's something we should strongly consider from now on.
+
+There are many benefits to including the version of Kani in its output.
+However, I think the main ones will be the following:
+ * **Earlier detection of version-related discrepancies**:
+ Users are likely to discuss discrepancies in verification outcomes by looking at Kani's output.
+ These may look exactly the same[^cbmc-version] (except for the discrepant value) on two different versions of Kani.
+ Including the version will help users realize sooner that they're using different versions of Kani.
+ * **Simpler issue triaging**:
+ New issues require users to post the Kani version they used.
+ Getting this information requires another call with `--version`, which wouldn't be needed if we simply printed the version.
+ Also, note that users may need to do more work if they aren't running Kani locally (e.g., Kani running in CI).
+
+In addition, printing the Kani version may be useful for other purposes (automate CI processes, help users realize they're using outdated versions, etc.).
+
+## User Experience
+
+The first line printed in any Kani invocation (either through `kani` or `cargo kani`, and regardless of subcommands) will inform users of the version.
+The behavior will be extended for development versions, where it'll print the short hash of the HEAD commit in addition to the version.
+
+### Release versions
+
+The first line to be printed will be:
+
+```
+Launching the Kani Rust Verifier <version>
+```
+
+where `<version>` is the version of Kani under use, which follows the semantic versioning format `MAJOR.MINOR.PATCH`.
+
+For example, for the release version of [Kani 0.29.0](https://github.com/model-checking/kani/releases/tag/kani-0.29.0), this would have printed:
+
+```
+Launching the Kani Rust Verifier 0.29.0
+```
+
+### Development versions
+
+The first line to be printed will be:
+
+```
+Launching the Kani Rust Verifier <version> (dev. version - commit: <commit>)
+```
+
+where `<version>` is the version of Kani under use, which follows the semantic versioning format `MAJOR.MINOR.PATCH`,
+and `<commit>` is the short hash (i.e., 7 hexadecimal digits with format `hhhhhhh`) of the `HEAD` commit.
```

Comment:
If we wanted to be super extra we could also print `(dirty)` if your worktree has uncommitted changes or add a hash for all files in `src` :joy: